### PR TITLE
RFR - Footer

### PR DESCRIPTION
**Pull request by dave315**

### CODE REVIEW

File: app/controllers/homes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class HomesController < ApplicationController
+  def index
+    if params[:search]
+      @search_result = Rubygem.where(
+        "name like ?", "%#{params[:search]}%"
+      )
+    end
+    @featuredgem = Rubygem.where(featured: true)
```
Comment (vikram7) [inline app/controllers/homes_controller.rb]: variable name should be plural `@featuredgems` and should also be ruby format: `@featured_gems`

File: app/views/homes/index.html.erb [hunk 2]
```diff
@@ -1,11 +1,46 @@
...
+    <% end %>
+  </div>
 
-<h1>Gemhunter</h1>
-
-<img src="http://ecx.images-amazon.com/images/I/31W7FDPBX5L._SY445_.jpg"height="100" width="100">
+  <div class="row">
+    <div class="small-12 small-centered columns text-center">
+      <h1>Featured Gems</h1>
+      <% @featuredgem.each do |rubygem| %>
```
Comment (vikram7) [inline app/views/homes/index.html.erb]: Update here per the controller.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -21,44 +20,42 @@
...
+</div>
 </body>
-<footer class="footer">
-  <div class="row full-width">
-    <div class="small-12 medium-3 large-4 columns">
-      <h4></h4>
-    </div>
+<footer>
+  <div class="footer">
+    <p><%=link_to "About", about_index_path%></p>
```
Comment (vikram7) [inline app/views/layouts/application.html.erb]: Global comment. You have a bunch of `<p>` and `<li>` tags which likely should be `<div>` tags especially as you start styling more.

File: spec/controllers/homes_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe HomesController, type: :controller do
+
+end
```
Comment (vikram7) [inline spec/controllers/homes_controller_spec.rb]: Does this controller test need to be there if you're not using it?

File: app/models/rubygem.rb [hunk 5]
```diff
@@ -5,4 +5,35 @@ class Rubygem < ActiveRecord::Base
...
+        "name like ?", "%#{params[:search]}%"
+      )
+    elsif params[:order_by] && !params[:search]
+      order_by_case(params)
+    elsif params[:search] && params[:order_by]
+      order_by_case(params)
+    else
+      Rubygem.all
+    end
+  end
```
Comment (davidrf) [inline app/models/rubygem.rb]: Yea this looks pretty horrible. So you want to do two things here. First if there is a search params, you want to get that list else you want to get all the ruby gems. Second, is you want to order the list if there is an order params.

```ruby
def self.filter(params)
  if params[:search]
    ruby_gems = where("name like ?", "%#{params[:search])%")
  else
    ruby_gems  = all
  end

  if params[:order_by]
    ruby_gems = order_by_case(ruby_gems, params[:order_by])
  end
  ruby_gems
end

```

File: app/models/rubygem.rb [hunk 6]
```diff
@@ -5,4 +5,35 @@ class Rubygem < ActiveRecord::Base
...
+    when "most_reviewed"
+      Rubygem.where(
+        "name like ?", "%#{params[:search]}%"
+      ).order(:review_count)
+    when "newest"
+      Rubygem.where(
+        "name like ?", "%#{params[:search]}%"
+      ).order(created_at: :desc)
+    end
+  end
```
Comment (davidrf) [inline app/models/rubygem.rb]: This is doing searches and order bys. It should only do one thing.

File: app/controllers/rubygems_controller.rb [hunk 7]
```diff
@@ -1,29 +1,9 @@
...
-            "name like ?", "%#{params[:search]}%"
-          ).order(:review_count)
-        when "newest"
-          Rubygem.where(
-            "name like ?", "%#{params[:search]}%"
-          ).order(created_at: :desc)
-        end
-    end
-    @featuredgem = Rubygem.where(featured: true)
+    @gem_searched = params[:search]
```
Comment (davidrf) [inline app/controllers/rubygems_controller.rb]: I would get rid of this and make this a helper method on the view

File: app/views/homes/root.html.erb [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      <div class="two columns">
+        <%= submit_tag("Search for Gems", class: "button expand postfix") %>
+      </div>
+    <% end %>
+  </div>
+
+  <div id="sort_by">
+      <a href="#" data-dropdown="drop" class="button tiny">Sort By &raquo;</a>
+      <ul id="drop" class="tiny content f-dropdown" data-dropdown-content>
+        <li> <%= link_to "Highest Rated", rubygems_path(search: @gem_searched, order_by: "highest_rated") %></li>
```
Comment (davidrf) [inline app/views/homes/root.html.erb]: I would make `@gem_searched` a view helper method that's just `gem_searched`

File: app/views/homes/root.html.erb [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        <li> <%= link_to "Most Reviewed", rubygems_path(search: @gem_searched, order_by: "most_reviewed") %></li>
+        <li> <%= link_to "Newest", rubygems_path(search: @gem_searched, order_by: "newest") %></li>
```
Comment (davidrf) [inline app/views/homes/root.html.erb]: Same comment for lines 20 and 21

**Pull request merged**